packages/graphql/src/schema/pagination.ts

```diff
@@ -17,17 +17,65 @@
  * limitations under the License.
  */
 
+import { FieldNode, GraphQLResolveInfo, SelectionSetNode } from "graphql";
 import { getOffsetWithDefault, offsetToCursor } from "graphql-relay/connection/arrayConnection";
 import { Integer, isInt } from "neo4j-driver";
+import { ConnectionField, ConnectionQueryArgs } from "../types";
+
+function getAliasKey({ selectionSet, key }: { selectionSet: SelectionSetNode | undefined; key: string }): string {
+    const selection = (selectionSet?.selections || []).find(
+        (x) => x.kind === "Field" && x.name.value === key
+    ) as FieldNode;
+
+    if (selection?.alias) {
+        return selection.alias.value;
+    }
+
+    return key;
+}
+
+export function connectionFieldResolver({
+    connectionField,
+    rootValue,
+    args,
+    resolveInfo,
+}: {
+    connectionField: ConnectionField;
+    rootValue: any;
+    args: ConnectionQueryArgs;
+    resolveInfo: GraphQLResolveInfo;
+}) {
+    const firstField = resolveInfo.fieldNodes[0];
+    const selectionSet = firstField.selectionSet;
+
+    let value = rootValue[connectionField.fieldName];
+    if (firstField.alias) {
+        value = rootValue[firstField.alias.value];
+    }
+
+    const totalCountKey = getAliasKey({ selectionSet, key: "totalCount" });
+    const totalCount = value.totalCount;
+
+    return {
+        [totalCountKey]: isInt(totalCount) ? totalCount.toNumber() : totalCount,
+        ...createConnectionWithEdgeProperties({ source: value, selectionSet, args, totalCount }),
+    };
+}
 
 /**
  * Adapted from graphql-relay-js ConnectionFromArraySlice
  */
-export function createConnectionWithEdgeProperties(
-    arraySlice: { node: Record<string, any>; [key: string]: any }[] = [],
-    args: { after?: string; first?: number } = {},
-    totalCount: number
-) {
+export function createConnectionWithEdgeProperties({
```

Comment:
I can't really argue with this as it obviously works... But it seems a "shame" that this is no longer really "Adapted from graphql-relay-js ConnectionFromArraySlice" as per the function comment.

Very open-ended question, was there no other way this could have been done whilst maintaining the original function? I'm just intrigued as to how the function from the `graphql-relay` package achieves aliasing, because it must do I imagine?